assets/js/services/adverts.service.js

```diff
@@ -21,16 +21,22 @@ function Adverts ($http, $rootScope) {
   }
 
   function fetch () {
-    let advertsFeed = $rootScope.rootURL + 'adverts_feed?wallet_version=3';
+    if (!$rootScope.apiDomain.endsWith('.blockchain.info/')) {
+      return;
+    }
+
+    let advertsFeed = $rootScope.apiDomain + 'ads/get?wallet=true&n=2';
     $http.get(advertsFeed)
-      .success(data => {
-        let adverts = data.partners.home_buttons.splice(0);
-        service.ads.push(randFromArray(adverts));
-        service.ads.push(randFromArray(adverts));
-      });
-  }
+      .success(function (data) {
+        data.forEach(function (ad) {
+          if (!/^data:image\/(png|jpg|jpeg|gif);base64,/.test(ad.data) ||
+              !angular.isNumber(ad.id) ||
+              !/^[0-9a-zA-Z ]*$/.test(ad.name)) {
```

Comment:
This might be too strict for non-latin alphabet advertisers, but can expand that later if it's not an issue yet.